src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -268,6 +232,54 @@ function toggleCollapse(e, pans) {
     }
 }
 
+/**
+ * Expand each panel. If the panel data is not loaded, load it by Ajax first before expansion.
+ */
+function expand(panels, needToCheckForAjaxLoading) {
+    var i = 0;
+    for (var idx = 0; idx < panels.length; idx++) {
+        if (needToCheckForAjaxLoading) {
+            // When the panel's parent element has class ajax_auto or ajax-response-auto, the panel data has
+            // not been loaded yet. We will need to load the panel data by Ajax.
+            var $ajaxAuto = $(panels[idx]).parent().children('.ajax_auto');
+            var $ajaxResponseAuto = $(panels[idx]).parent().children('.ajax-response-auto');
+            var hasAjaxAutoLoading = $ajaxAuto.length !== 0;
+            var hasAjaxResponseAutoLoading = $ajaxResponseAuto.length !== 0;
+
+            // Ajax loading is triggered by clicking on the element. When clicked, the panel data is loaded
+            // by Ajax, and ajax_auto or ajax-response-auto class will be removed from the element.
+            if (hasAjaxAutoLoading) {
+                $ajaxAuto.click();
+            }
+            if (hasAjaxResponseAutoLoading) {
+                $ajaxResponseAuto.click();
+            }
+            if (hasAjaxAutoLoading || hasAjaxResponseAutoLoading) {
+                // When the element needs Ajax loading, it must be collapsed because all expanded panels are
```

Comment:
hmm.. would it be better to say that `Prior to Ajax load, element is collapsed because data is not loaded yet. Clicking toggles both loading of data and expansion of element.`?

Also should it be `element` or `panel`?